src/Common/Editor/Full_Site/Template_Utils.php

```diff
@@ -85,4 +90,123 @@ public static function inject_theme_attribute_in_content( $template_content ) {
 
 		return $template_content;
 	}
+
+	/**
+	 * Fetch a WP_Block_Template by the post_name and terms field.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $post_name The post_name field to search by.
+	 * @param string $terms     The terms field to search by, default to 'tec'.
+	 *
+	 * @return WP_Block_Template|null The newly created WP_Block_Template, or null on error.
+	 */
+	public static function find_block_template_by_post( string $post_name, string $terms = 'tec' ): ?WP_Block_Template {
+		// Let's see if we have a saved template?
+		$wp_query_args  = [
+			'post_name__in'  => [ $post_name ],
+			'post_type'      => 'wp_template',
+			'post_status'    => [ 'auto-draft', 'draft', 'publish', 'trash' ],
+			'posts_per_page' => 1,
+			'no_found_rows'  => true,
+			'tax_query'      => [
+				[
+					'taxonomy' => 'wp_theme',
+					'field'    => 'name',
+					'terms'    => $terms,
+				],
+			],
+		];
+		$template_query = new WP_Query( $wp_query_args );
+		$posts          = $template_query->posts;
+
+		// If empty, this is our first time loading our Block Template. Let's create it.
```

Comment:
But we don't create it here. Perhaps this needs to be reworded?

Reply:
Good catch, I guess from old logic.

Reply:
According to the method docblock, `null` means an error.

Reply:
Yeah, figured it might be C&P or typing from memory. No big deal, thanks!